### ISSUE TITLE
Plasma completely freezes on Libreoffice launch under wayland

### ISSUE THREAD

**Issue by celiopy** [paulmcauley/klassy#68]
Desktop freezes when I run libreoffice (calc, writer...) while under Wayland and I have to force shutdown laptop to leave. Didn't notice this behaviour on X11. Anyone ever faced it?

**Comment by TheXplodR** [comment]
I have similar issue but with Kate (I didn't tried with LibreOffice). The laptop still works on other tty-s (openSUSE Tumbleweed 20221218). On X11 the issue is not present.

**Comment by luisbocanegra** [comment]
Cannot reproduce it here, Archlinux with Plasma (wayland) from master, could it be something in your  klassyrc configuration?

**Comment by KH-trapezoide** [comment]
Same problem on Arch and EndeavourOS, after using Klassy I have random freezes when I open some programs. Gwenview has a 100% chance of crashing on Plasma 5.26 (but I had this issue since 5.23 at least).
The problem dissapears after using another theme.

**Comment by aquova** [comment]
I stopped using klassy some months ago for this exact reason; I was curious if it had been addressed. Lots of freezes opening very common programs, I recall gwenview and okular would almost always cause freezes.

**Comment by luisbocanegra** [comment]
Please share your system details, are you using both klassy's window decoration & application theme, Kvantum, NVIDIA, AMD??

Also share a log of the crash (guide [here](https://community.kde.org/Guidelines_and_HOWTOs/Debugging/How_to_create_useful_crash_reports)) and ~/.config/klassyrc so others can try to reproduce the issue.

**Comment by celiopy** [comment]
As I haven't been using it for quite a while now, I can't actually provide the logs. The issue only happens when over wayland, and it appears to be related. Klassy works well while on xorg, as the applications I mentioned works well. I was using arch linux with plasma, my laptop is an "old" 8th gen Intel I5, 8gb ram. ssd. No discrete graphics card.

**Comment by aquova** [comment]
I was able to reinstall and almost immediately reproduce the issue just now on my Framework laptop. 

Steps to reproduce:
1. Installed the AUR `klassy-git` package
2. In KDE Plasma settings, set Appearance -> Window Decorations to Klassy.
3. Opened up Okular, the system immediately hard froze. I did try some other applications, like Gwenview or reopening the KDE settings, but those behaved fine.

Looking through my journalctl output, there isn't much of interest. These are the only entries between opening Okular and me forcing the computer to shut down by hard pressing the power button. If there are any additional logs people would like gathered, I can do so.

```
Jan 12 08:13:17 neptune systemd[1287]: Started Okular - Document Viewer.
Jan 12 08:13:18 neptune kded5[1413]: org.kde.plasma.appmenu: Got an error
Jan 12 08:13:18 neptune kded5[1413]: org.kde.plasma.appmenu: Got an error
Jan 12 08:13:18 neptune kwin_wayland[1320]: kwin_wayland_drm: an error occurred while swapping buffers "EGL_BAD_SURFACE"
Jan 12 08:13:31 neptune systemd-logind[517]: Power key pressed short.
```

System:
OS: Arch Linux
CPU: Intel i7-1165G7
GPU: Intel TigerLake-LP GT2
DE: KDE Plasma 5.26.5 Wayland

Note that I'm almost certain I've seen this on a Ryzen/Radeon machine as well, but I didn't try to reproduce it there.

My klassyrc file is shown below. I've never modified this directly, only via the GUI in the KDE settings application.
```
[Common]
BackgroundColors=ColorsAccentWithTrafficLights
ButtonIconStyle=StyleKite
DrawTitleBarSeparator=true

[Default Windeco Exception 0]
BorderSize=1
Enabled=true
ExceptionProgramNamePattern=kdenlive
ExceptionWindowPropertyPattern=.*Kdenlive
ExceptionWindowPropertyType=1
HideTitleBar=false
Mask=0
OpaqueTitleBar=true
PreventApplyOpacityToHeader=true

[Windeco]
CornerRadius=2
```

**Comment by luisbocanegra** [comment]
What about dmesg?

**Comment by celiopy** [comment]
> ```
> Jan 12 08:13:17 neptune systemd[1287]: Started Okular - Document Viewer.
> Jan 12 08:13:18 neptune kded5[1413]: org.kde.plasma.appmenu: Got an error
> Jan 12 08:13:18 neptune kded5[1413]: org.kde.plasma.appmenu: Got an error
> Jan 12 08:13:18 neptune kwin_wayland[1320]: kwin_wayland_drm: an error occurred while swapping buffers "EGL_BAD_SURFACE"
> Jan 12 08:13:31 neptune systemd-logind[517]: Power key pressed short.
> ```

That shows that it is related to wayland, right?

**Comment by aquova** [comment]
> What about dmesg?

Nothing for a few minutes prior to the event.

However, over lunch I went and reproduced it on my desktop machine with the same procedure, and got slightly different results. Here, only my primary monitor froze (and has maintained frozen while I write this) while my secondary monitor is fine. Here are the relevant journalctl entries:

```
Jan 12 12:37:13 mars systemd[1276]: Started Okular - Document Viewer.
Jan 12 12:37:13 mars kded5[1410]: org.kde.plasma.appmenu: Got an error
Jan 12 12:37:13 mars kded5[1410]: org.kde.plasma.appmenu: Got an error
Jan 12 12:37:13 mars kwin_wayland[1313]: kwin_wayland_drm: an error occurred while swapping buffers "EGL_BAD_SURFACE"
Jan 12 12:37:23 mars plasmashell[1468]: org.kde.plasma.libtaskmanager: Got invalid activation app_id: "org.kde.krunner"
Jan 12 12:37:23 mars krunner[2364]: QCommandLineParser: argument list cannot be empty, it should contain at least the executable name
Jan 12 12:37:23 mars krunner[2364]: kf.sonnet.core: No language dictionaries for the language: "en_US"
Jan 12 12:37:23 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:23 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:23 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:23 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:23 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:23 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:23 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:23 mars krunner[2364]: qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
Jan 12 12:37:24 mars systemd[1276]: Started Okular - Document Viewer.
Jan 12 12:37:24 mars kded5[1410]: org.kde.plasma.appmenu: Got an error
Jan 12 12:37:24 mars kded5[1410]: org.kde.plasma.appmenu: Got an error
Jan 12 12:37:34 mars plasmashell[1468]: org.kde.plasma.libtaskmanager: Got invalid activation app_id: "org.kde.krunner"
Jan 12 12:37:34 mars krunner[2364]: QCommandLineParser: argument list cannot be empty, it should contain at least the executable name
Jan 12 12:37:34 mars krunner[2364]: kf.sonnet.core: No language dictionaries for the language: "en_US"
Jan 12 12:37:34 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:34 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:34 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:34 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:34 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:34 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:34 mars krunner[2364]: kf.plasma.quick: Couldn't create KWindowShadow for View(0x7fffdadc92d0)
Jan 12 12:37:34 mars krunner[2364]: qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
Jan 12 12:37:36 mars kstart5[28338]: Omitting both --window and --windowclass arguments is not recommended
```

You can see that I ran Okular twice. The first time I opened it, the primary monitor froze, but the second monitor was okay. The second time, nothing seems to have happened (although my monitor is still frozen).

System details:
OS: Arch Linux
DE: KDE Plasma 5.26.5 Wayland
CPU: Ryzen 5 5600X
GPU: Radeon 6650XT

**Comment by d-avery** [comment]
I have this issue as well, but with Notepadqq as well, not just Libreoffice. Arch Linux, kernel 6.1.9-arch1-1. KDE Wayland 5.26.5 and 5.26.90 both affected.

**Comment by paulmcauley** [comment]
This is a duplicate of #53.

I have released Klassy version 4.1 with a workaround for this. I have tested it on Plasma 5.27.4. Please try 4.1 and report back on whether this issue is still present or not.

The EGL_BAD_SURFACE is caused by a paletteUpdated signal triggering the shadow to update (particularly on opening application with custom colour schemes). If the shadow tries to update at the same time as the paint() method then the EGL_BAD_SURFACE (and segfault in Plasma 5.26+) occurs.

This workaround abandons the attempt to update the shadow if the paint() method is executing. The workaround is not ideal though, and I think that it might be more elegant to have a solution in kwin.
